File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/abstract_factory/lab/conventional_tomato.cpp
// conventional_tomato.cpp

#include "conventional_tomato.h"
#include "tomato.h"

conventional_tomato::conventional_tomato(double a_price, int a_number_of_pesticides)
: tomato(a_price)
{
	my_number_of_pesticides = a_number_of_pesticides;
}

conventional_tomato::~conventional_tomato()
{
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/singleton/abstract_factory/lab/ConventionalTomato.java
// ConventionalTomato.java

public class ConventionalTomato extends Tomato
{
	public ConventionalTomato(double aPrice, int aNumberOfPesticides)
	{
		super(aPrice);
		myNumberOfPesticides = aNumberOfPesticides;
	}

	public ConventionalTomato(double aPrice)
	{
		super(aPrice);
		myNumberOfPesticides = 0;
	}

	private int myNumberOfPesticides;
}
